### ISSUE TITLE
Forwarding multiple arguments to method reference fails

### ISSUE THREAD

**Issue by okirmis** [oracle/truffleruby#1629]
Forwarding multiple arguments to a method reference works in MRI, but fails in truffleruby. Example:

```ruby
def two_argument_method(key, value)
  puts "#{key}: #{value}"
end

hash = { a: 1 }
hash.each(&method(:two_argument_method))
```

On MRI it prints `a: 1` to the command line. Truffleruby fails with:
```
minimal_example.rb:1:in `two_argument_method': wrong number of arguments (given 1, expected 2) (ArgumentError)
        from minimal_example.rb:1:in `two_argument_method'
        from minimal_example.rb:6:in `each'
        from minimal_example.rb:6:in `<main>'
```

`Hash#each` passes two arguments. Forwarding single arguments, e.g. with `Array#each` works fine.

```
> ruby --version
truffleruby 1.0.0-rc10, like ruby 2.4.4, GraalVM CE Native [x86_64-linux]
```

Edit: For some reason it pretends to be rc10, but I used `rvm install truffleruby-1.0.0-rc14` and `rvm use...`. RVM said `Found remote file https://github.com/oracle/truffleruby/releases/download/vm-1.0.0-rc14/truffleruby-1.0.0-rc14-linux-amd64.tar.gz` so I believe it's actually rc14.

**Comment by eregon** [comment]
I would trust the `ruby --version` output, this is probably a RVM bug, see https://github.com/rvm/rvm/issues/4633

Anyway, the given example still fails on master, so the version doesn't matter for this bug report.

**Comment by eregon** [comment]
This seems to be a bug of `Hash#each` rather, which currently yields an Array of 2 elements instead of yielding 2 arguments:
```ruby
def two_argument_method(key, value)
  puts "#{key}: #{value}"
end

def y
  yield 1, 2
end

y(&-> a,b { p [a,b] })
y(&method(:two_argument_method))

hash = { a: 1 }
hash.each(&-> a,b { p [a,b] })
hash.each(&method(:two_argument_method))
```
On TruffleRuby master:
```
[1, 2]
1: 2
methargs.rb:14:in `block in <main>': wrong number of arguments (given 1, expected 2) (ArgumentError)
	from methargs.rb:14:in `each'
	from methargs.rb:14:in `<main>'
```

**Comment by eregon** [comment]
@okirmis Fix for this coming.

I wouldn't recommend using Method or lambdas with yielding methods from Array/Hash in general, because it's quite inconsistent what works and what doesn't. Better to use a block and delegate explicitly, that always works no matter the core method used.
In this case, there is no way to pass all specs with a single choice between yielding 2 arguments or an Array of 2 elements, so MRI actually introduced a special case just to make this work (maybe for compatibility?):
https://github.com/ruby/ruby/blob/638c96858274deb90b2b923477d6b9525e1ed065/hash.c#L2870-L2878

**Comment by eregon** [comment]
@okirmis Thank you for the report. I fixed this in 7c7997b02780494761ec5a51f3d30577b8c61db9 and it will be in the next release, 1.0.0-rc15.
